### PR TITLE
Add default AppCatalog to kubermatic-installer

### PR DESCRIPTION
**Pull request by SimonTheLeg**
**What this PR does / why we need it**:

This PR adds the implementation for the default AppCatalog to the kubermatic installer via a flag. Currently it is just pointing to a repo with some sample AppDefs, but as soon as the final place for the official catalog AppDefs is found, it can easily be swapped in.

**Which issue(s) this PR fixes**:

Part of #12527  

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Default ApplicationCatalog can be deployed via --deploy-default-app-catalog flag (EE only)
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: cmd/kubermatic-installer/cmd_deploy.go [hunk 1]
```diff
@@ -142,6 +144,8 @@ func DeployCommand(logger *logrus.Logger, versions kubermaticversion.Versions) *
 	cmd.PersistentFlags().BoolVar(&opt.MLAForceMLASecrets, "mla-force-secrets", false, "(UserCluster MLA) force reinstallation of mla-secrets Helm chart")
 	cmd.PersistentFlags().BoolVar(&opt.MLAIncludeIap, "mla-include-iap", false, "(UserCluster MLA) Include Identity-Aware Proxy installation")
 
+	cmd.PersistentFlags().BoolVar(&opt.DeployDefaultAppCatalog, "deploy-default-app-catalog", false, "Deploy the default Application Catalog (EE only)")
```
Comment (xrstf) [inline cmd/kubermatic-installer/cmd_deploy.go]: If this is EE only, please define a wrapper in the main_ce/main_ee that returns the persistent flags, with CE returning [] and EE returning this one flag.

File: cmd/kubermatic-installer/cmd_deploy.go [hunk 2]
```diff
@@ -186,6 +190,11 @@ func DeployFunc(logger *logrus.Logger, versions kubermaticversion.Versions, opt
 			)
 		}
 
+		// default App Catalog feature is only available for EE, so error out early if being used with CE
```
Comment (xrstf) [inline cmd/kubermatic-installer/cmd_deploy.go]: We use compile-time settings, never runtime checks for the edition.

File: cmd/kubermatic-installer/cmd_deploy.go (hunk 1 above)
Comment (SimonTheLeg) [reply cmd/kubermatic-installer/cmd_deploy.go]: I have pushed how I understood your idea to [`9b298b2` (#12623)](https://github.com/kubermatic/kubermatic/pull/12623/commits/9b298b221da3010684e9d6c43eaa63394f5dc61e). ptal if that is how you envisioned the code to be structured.

File: cmd/kubermatic-installer/cmd_deploy.go (hunk 2 above)
Comment (SimonTheLeg) [reply cmd/kubermatic-installer/cmd_deploy.go]: ok. I guess I then the only way I could think of is to create a stack_ce.go and a stack_ee.go which then get called in [pkg/install/stack/kubermatic-master/stack.go](https://github.com/kubermatic/kubermatic/pull/12623/files#diff-eea3bf1b32bfc7d0d71cba02976c90303e11a71e008542896f99fa4ffdfc2466). The stack_ce version is essentially a no-op. Would this be our preferred approach?

**Comment by SimonTheLeg** [comment]
I have now added the appdefs into the kubermatic repository. At this point I am unsure whether to place them inside
`cmd/kubermatic-installer` or rather `pkg/stack/kubermatic-master` or in a completely different place. Any strong opinions?

**Comment by xrstf** [comment]
To unblock this, I decree a place somewhere in `pkg/`, just like we do for our CRDs (pkg/crd/). :grin:

**Comment by SimonTheLeg** [comment]
made most of the required changes. @xrstf could you have a look at my previous responses on your review comments? There are still some things I am not 100% sure what our preferred approach is.

File: pkg/install/stack/kubermatic-master/default_application_catalog.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
```
Comment (xrstf) [inline pkg/install/stack/kubermatic-master/default_application_catalog.go]: EE code has to live in pkg/ee/. Just like the main_ce/ee, please create a wrappers_ce/ee and either call code in pkg/ee/ or do a NOP.

File: cmd/kubermatic-installer/cmd_deploy.go (hunk 2 above)
Comment (xrstf) [reply cmd/kubermatic-installer/cmd_deploy.go]: Yep, exactly. Similar to https://github.com/kubermatic/kubermatic/blob/main/cmd/master-controller-manager/wrappers_ce.go#L33-L40

File: cmd/kubermatic-installer/cmd_deploy.go (hunk 1 above)
Comment (xrstf) [reply cmd/kubermatic-installer/cmd_deploy.go]: Exactly. :)

**Comment by SimonTheLeg** [comment]
/unhold

ready for merging

**Review by xrstf — APPROVED** [review]
Beautiful

/approve

**Pull request merged**